### DISCUSSION TITLE
Boundaries for FR and CDR regions

### DISCUSSION THREAD

**Discussion by y1zhou** [milaboratory/mixcr#832] [Q&A] [answered]
In [IMGT unique numbering for V-DOMAIN and V-LIKE-DOMAIN](https://www.imgt.org/IMGTScientificChart/Numbering/IMGTIGVLsuperfamily.html), the following bounds were given:

> The IMGT unique numbering provides a standardized delimitation of the framework regions (FR1-IMGT: positions 1 to 26, FR2-IMGT: 39 to 55, FR3-IMGT: 66 to 104 and FR4-IMGT: 118 to 128) and of the complementarity determining regions: CDR1-IMGT: 27 to 38, CDR2-IMGT: 56 to 65 and CDR3-IMGT: 105 to 117.

The `anchorPointPositions` seem to be positions for nucleotide sequences, so the numbers above should be multiplied by 3. These positions still don't seem to match with the bounds given in the code:

https://github.com/milaboratory/mixcr/blob/7a85aa610f35a162e0ac58b4700a1c6c937db297/src/main/resources/parameters/segment_importer_parameters.json#L17-L46

I tested feeding a heavy chain sequence generated from MixCR into [ANARCI](https://github.com/oxpig/ANARCI). the MixCR CDR3 had one additional amino acid on both sides when compared to the ANARCI IMGT-scheme labeled CDR3.

**Comment by Alex-Davydov** [comment, marked as answer]
Hi,
These parameters are not used, now all coordinates for every species from IMGT are stored in this repository:
[https://github.com/repseqio/library-imgt/tree/master/rules](url) (pay attention to 0 based indexing in contrast to IMGT where coordinates start from 1).
Concerning the CDR3 and conservative cysteine and tryptophan / phenylalanine:
The coordinates of frameworks and CDRs in IMGT are determend by protein structure, so the conservative cysteine ​​belongs to FR3 and CDR3 begins after it. But there are situations when 1 or 2 letters are cut off from the cysteine ​​triplet during VDJ recombination, and in order not to lose these sequences, cysteine ​​is included in CDR3 in MiXCR (same for conservative tryptophan / phenylalanine in the J gene).
